pandas/core/internals/managers.py

```diff
@@ -571,12 +574,17 @@ def replace_list(self, src_list, dest_list, inplace=False, regex=False,
         # figure out our mask a-priori to avoid repeated replacements
         values = self.as_array()
 
-        def comp(s):
+        def comp(s, reg=False):
             if isna(s):
                 return isna(values)
-            return _maybe_compare(values, getattr(s, 'asm8', s), operator.eq)
+            if hasattr(s, 'asm8'):
+                return _maybe_compare(maybe_convert_objects(values),
+                                      getattr(s, 'asm8'), reg)
+            if reg and is_re_compilable(s):
```

Comment:
My mistake. At first, I wanted to raise a ValueError when regex is True but replacer is not reg compilable. It might not be a good idea to do it here. I will delete the if condition.